pkgs/applications/graphics/pinta/default.nix

```diff
@@ -37,35 +38,45 @@ buildDotnetModule rec {
     owner = "PintaProject";
     repo = "Pinta";
     rev = version;
-    sha256 = "sha256-iOKJPB2bI/GjeDxzG7r6ew7SGIzgrJTcRXhEYzOpC9k=";
+    sha256 = "sha256-Bvzs1beq7I1+10w9pmMePqGCz2TPDp5UK5Wa9hbKERU=";
   };
 
-  postFixup = ''
-    # Rename the binary
-    mv $out/bin/Pinta $out/bin/pinta
-
-    # Copy desktop icons
-    for size in 16x16 22x22 24x24 32x32 96x96 scalable; do
-      mkdir -p $out/share/icons/hicolor/$size/apps
-      cp xdg/$size/* $out/share/icons/hicolor/$size/apps/
-    done
-
-    # Copy runtime icons
-    cp -r Pinta.Resources/icons/hicolor/16x16/* $out/share/icons/hicolor/16x16/
+  # https://github.com/NixOS/nixpkgs/issues/38991
+  # bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
+  LOCALE_ARCHIVE = "${glibcLocales}/lib/locale/locale-archive";
 
-    # Install manpage
-    installManPage xdg/pinta.1
+  # Do the autoreconf/Makefile job manually
+  # TODO: use upstream build system
+  postFixup = ''
+    # Substitute translation placeholders
+    substitute xdg/pinta.appdata.xml.in xdg/pinta.appdata.xml \
+      --replace _p p
 
-    # Fix and copy desktop file
-    # TODO: fix this propely by using the autoreconf+pkg-config build system
-    # from upstream
-    mkdir -p $out/share/applications
-    substitute xdg/pinta.desktop.in $out/share/applications/Pinta.desktop \
+    substitute xdg/pinta.desktop.in xdg/pinta.desktop \
       --replace _Name Name \
       --replace _Comment Comment \
       --replace _GenericName GenericName \
       --replace _X-GNOME-FullName X-GNOME-FullName \
       --replace _Keywords Keywords
+
+    # Rename binary
+    mv $out/bin/Pinta $out/bin/pinta
+
+    # Build and copy translations
+    # https://github.com/PintaProject/Pinta/blob/8b1bcd8802ecdfae1cb85683d1574c26c23b6d6e/Pinta/Pinta.csproj#L35-L39
+    for po in po/*.po; do
+      language="$(basename $po .po)"
+      translation_dir="publish/locale/$language/LC_MESSAGES"
+      mkdir -p "$translation_dir"
+      msgfmt "$po" -o "$translation_dir/pinta.mo"
+    done
```

Comment:
I think there is a better way to do this. I got it to compile without errors using this instead:

```
    dotnet msbuild Pinta/Pinta.csproj -target:CompileTranslations
    dotnet msbuild Pinta/Pinta.csproj -target:PublishTranslations
```

However the result was not available on `/build/source/publish` directory. Also, maybe there is a better way to do this that doesn't involve `postFixup`. Maybe I can use `dotnetBuildFlags` instead or something?

CC @IvarWithoutBones.

Reply:
Fixed it on b9939ef.